Providers/Profile.swift

```diff
@@ -1235,8 +1235,8 @@ open class BrowserProfile: Profile {
                         engineResults: results,
                         stats: statsSession.hasStarted() ? statsSession.end() : nil
                     )
-                    gleanHelper.end(result)
                     self.endSyncing(result)
+                    gleanHelper.end(result)
```

Comment:
This change was needed since GleanTelemetryTests.testSyncPingIsSentOnSyncOperation() test teardown wasl called before endSyncing was done and was resulting in a crash (due to `defer` being filled after the end of the test). Some other solutions would have been to wait for completion with a closure so our test would wait, or having mocks so we would control actions in this `syncSeveral` method. I instead opted to change when the glean ping is called. Effect of this should only be that the Glean telemetry ping is sent a little bit after.